conda_build/metadata.py

```diff
@@ -131,227 +138,58 @@ def remove_constructor(cls, tag):
-            log.debug(
-                "Treating unknown selector '" + missing_var + "' as if it was False."
-            )
-        next_string = selector_string.replace(missing_var, "False")
-        return eval_selector(next_string, namespace, variants_in_place)
-
-
-@lru_cache(maxsize=None)
-def _split_line_selector(text: str) -> tuple[tuple[str | None, str], ...]:
-    lines: list[tuple[str | None, str]] = []
-    for line in text.splitlines():
-        line = line.rstrip()
-
-        # skip comment lines, include a blank line as a placeholder
-        if line.lstrip().startswith("#"):
-            lines.append((None, ""))
-            continue
-
-        # include blank lines
-        if not line:
-            lines.append((None, ""))
-            continue
-
-        # user may have quoted entire line to make YAML happy
-        trailing_quote = ""
-        if line and line[-1] in ("'", '"'):
-            trailing_quote = line[-1]
-
-        # Checking for "[" and "]" before regex matching every line is a bit faster.
-        if (
-            ("[" in line and "]" in line)
-            and (match := sel_pat.match(line))
-            and (selector := match.group(3))
-        ):
-            # found a selector
-            lines.append((selector, (match.group(1) + trailing_quote).rstrip()))
-        else:
-            # no selector found
-            lines.append((None, line))
-    return tuple(lines)
-
-
-def select_lines(text: str, namespace: dict[str, Any], variants_in_place: bool) -> str:
-    lines = []
-    selector_cache: dict[str, bool] = {}
-    for i, (selector, line) in enumerate(_split_line_selector(text)):
-        if not selector:
-            # no selector? include line as is
-            lines.append(line)
-        else:
-            # include lines with a selector that evaluates to True
-            try:
-                if selector_cache[selector]:
-                    lines.append(line)
-            except KeyError:
-                # KeyError: cache miss
-                try:
-                    value = bool(eval_selector(selector, namespace, variants_in_place))
-                    selector_cache[selector] = value
-                    if value:
-                        lines.append(line)
-                except Exception as e:
-                    raise CondaBuildUserError(
-                        f"Invalid selector in meta.yaml line {i + 1}:\n"
-                        f"offending selector:\n"
-                        f"  [{selector}]\n"
-                        f"exception:\n"
-                        f"  {e.__class__.__name__}: {e}\n"
-                    )
-    return "\n".join(lines) + "\n"
+deprecated.constant(
+    "24.11",
+    "25.1",
+    "get_selectors",
+    _get_selectors,
+    addendum="Use `conda_build.selectors.get_selectors` instead.",
+)
+deprecated.constant(
+    "24.11",
+    "25.1",
+    "ns_cfg",
+    _get_selectors,
+    addendum="Use `conda_build.selectors.get_selectors` instead.",
```

Comment:
```suggestion
    addendum="Use `conda_build.selectors.ns_cfg` instead.",
```

Reply:
`ns_cfg` was renamed to `get_selectors` in the new `conda_build.selectors` module

Reply:
oh woops! Sorry!